### PR TITLE
Update to Fedora 36 and other improvements

### PR DESCRIPTION
**Pull request by piegamesde**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -15,8 +15,8 @@ jobs:
       matrix:
         runtime:
           - name: gtk-latest
```
Comment (MGlolenstine) [inline .github/workflows/docker.yml:17]: I see how `gtk-latest` is confusing and seeing that builds from `main` will pretty much never succeed, I think we can just remove the `gtk-latest` and move the changed configuration into `gtk-4.6`.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -23,39 +23,33 @@ jobs:
...
+          password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Extract metadata (tags, labels) for Docker
         id: meta
         uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
         with:
           images: |
-            mglolenstine/gtk4-cross
-            ghcr.io/mglolenstine/gtk4-cross
+            ghcr.io/piegamesde/gtk4-cross
```
Comment (MGlolenstine) [inline .github/workflows/docker.yml:44]: Should be reverted before merged.

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -68,8 +62,7 @@ jobs:
...
```
Comment (MGlolenstine) [inline .github/workflows/docker.yml:65]: Should be reverted before merged.

File: gtk4-cross-base-hub/Dockerfile [hunk 4]
```diff
@@ -15,7 +15,7 @@ RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
 # Set the rust linker to gcc
 # RUN echo "[target.x86_64-pc-windows-gnu]\nlinker = \"x86_64-w64-mingw32-gcc\"\nar = \"x86_64-w64-mingw32-gcc-ar\"" > /home/rust/.cargo/config
 RUN dnf install -y gcc
-RUN . ~/.cargo/env && cargo install --git https://github.com/MGlolenstine/peldd_dependency_scanner.git
+RUN . ~/.cargo/env && cargo install --git https://github.com/piegamesde/peldd_dependency_scanner.git
```
Comment (MGlolenstine) [inline gtk4-cross-base-hub/Dockerfile:18]: Can be reverted once the PR is merged on `pelld_dependency_scanner`.

**Review by MGlolenstine — COMMENTED** [review]
A quick look around.

**Review by aknarts — APPROVED** [review]

**Comment by piegamesde** [comment]
We might want to include the following optional dependencies by default:

- `rust-gio-devel`, `intltool`: Build script support for `gio::compile_resources`
-  `adwaita-icon-theme` or `mingw64-adwaita-icon-theme`: To copy the libadwaita icons over for packaging

**Comment by MGlolenstine** [comment]
I'm sorry for taking ages on this.
I've just came around to testing it and it LGTM.

I would be up for adding those optional dependencies to the default, as the icons should be present.

**Review by MGlolenstine — APPROVED** [review]

**Pull request merged**